src/libraries/Microsoft.Extensions.Caching.Memory/tests/TokenExpirationTests.cs

```diff
@@ -207,6 +207,29 @@ public void TokenExpiresOnRegister()
             Assert.Null(result);
         }
 
+        [Fact]
+        public async Task PostEvictionCallbacksGetInvokedWhenMemoryCacheEntriesExpireWithAnActiveChangeToken()
+        {
+            using var cache = new MemoryCache(new MemoryCacheOptions());
+            var key = new object();
+
+            var cts = new CancellationTokenSource(TimeSpan.FromSeconds(1));
+            var tcs = new TaskCompletionSource<object>();
+
+            cache.Set(key, new object(), new MemoryCacheEntryOptions
+            {
+                ExpirationTokens = { new CancellationChangeToken(cts.Token) },
+                PostEvictionCallbacks = { new PostEvictionCallbackRegistration {
+                    EvictionCallback = (key, value, reason, state) => tcs.TrySetResult(new object()) } },
+            });
+
+            Assert.True(cache.TryGetValue(key, out _));
+
+            await tcs.Task.TimeoutAfter(TimeSpan.FromSeconds(10));
```

Comment:
Do we need a new dependency on TaskTimeoutExtensions? Instead, can we use a `ManualResetEvent` and verify `WaitOne` returns `true`?

Reply:
Here's what I was thinking:

```C#
            using var cache = new MemoryCache(new MemoryCacheOptions());
            var key = new object();

            var cts = new CancellationTokenSource();
            var callbackInvoked = new ManualResetEvent(false);

            cache.Set(key, new object(), new MemoryCacheEntryOptions
            {
                ExpirationTokens = { new CancellationChangeToken(cts.Token) },
                PostEvictionCallbacks = { new PostEvictionCallbackRegistration {
                    EvictionCallback = (key, value, reason, state) => callbackInvoked.Set() } },
            });

            Assert.True(cache.TryGetValue(key, out _));

            cts.Cancel();

            Assert.True(callbackInvoked.WaitOne(TimeSpan.FromSeconds(10)));
            Assert.False(cache.TryGetValue(key, out _));
```